src/packet/Packet_TileSectionSend.res

```diff
@@ -868,10 +882,12 @@ module Encode = {
       ->BufferWriter.packInt16(self.height)
 
     let lastTile = ref(None)
-    for y in 0 to self.height - 1 {
-      for x in 0 to self.width - 1 {
-        let tile = (self.tiles[y]->Option.getUnsafe)[x]->Option.getUnsafe
-        writer->decidePackTile(lastTile, tile)
+    for y in 0 to self.height {
+      for x in 0 to self.width {
+        if (y, x) != (self.height - 1, self.width - 1) {
+          let tile = (self.tiles[y]->Option.getUnsafe)[x]->Option.getUnsafe
+          writer->decidePackTile(lastTile, tile)
+        }
```

Comment:
What is this change?

Reply:
The for loop is inclusive at both ends in this language so 0 to self.height actually means the last iteration is y=self.height which is out of bounds. That's why it has to be self.height - 1
And I don't think its necessary to skip the last tile. If decidePackTile runs on the last one and is successful, then lastTile will be None and so the code below it won't need to do anything. If it isn't successful then the code below it will pack that tile.
It has been some time since I looked  at this section but that's the way I understand it